### PR TITLE
Add elasticsearch 7 support

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: content/docs/next-release/deployment.md [hunk 1]
```diff
@@ -277,14 +277,74 @@ docker run \
...
+Elasticsearch 6.8 supports 7.x, 6.x, 5.x compatible mappings. The upgrade has to be done
+first to ES 6.8, then apply data migration or wait until old daily indices are removed (this requires
+to start Jaeger with `--es.version=7` to force using ES 7.x mappings for newly created indices).
+
+Jaeger by default uses Elasticsearch ping endpoint (`/`) to derive the version which is used
+for index mappings selection. The version can be overridden by flag `--es.version`.
+
+##### Data migration
+
+The data migration requires to be done on Elasticsearch 6.8.
```
Comment (objectiser) [inline content/docs/next-release/deployment.md]: "requires to" -> "must"

File: content/docs/next-release/deployment.md [hunk 2]
```diff
@@ -277,14 +277,74 @@ docker run \
...
+curl -ivX POST -H "Content-Type: application/json" http://localhost:9200/_reindex -d @reindex.json
+{
+  "source": {
+    "index": "jaeger-span-*"
+  },
+  "dest": {
+    "index": "jaeger-span"
+  },
+  "script": {
+    "lang": "painless",
```
Comment (objectiser) [inline content/docs/next-release/deployment.md]: What does "painless" relate to?

Comment (pavolloffay) [reply content/docs/next-release/deployment.md]: It's the name of the scripting language https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-scripting-painless.html

Comment (objectiser) [reply content/docs/next-release/deployment.md]: ok thanks.

**Pull request merged**